tests/insert_license_test.py

```diff
@@ -48,6 +49,10 @@
             ('module_without_license.py', '#', 'module_with_license_nospace.py', True, ['--no-space-in-comment-prefix']),
             ('module_without_license.php', '/*| *| */', 'module_with_license.php', True, ['--insert-license-after-regex', '^<\\?php$']),
             ('module_without_license.py', '#', 'module_with_license_noeol.py', True, ['--no-extra-eol']),
+
+            ('module_without_license.groovy', '//', 'module_with_license.groovy', True, ['--flexible-years']),
```

Comment:
👍 Nice new tests!
Could you please add one with a badly formatted year range?

Reply:
Sure can. Do you have any thoughts on what issues in formatting we want to detect and how to react? 
I guess when spotting a range with additional spaces like `2015 - 2016` we should still be able to upgrade the year. 
With `2015-16` should we log an error and skip modification?
Another thing is I assumed there's only one line with years, so if there are multiple just the first one will get modified. I don't think it's a common case so I didn't support it. There are many negative cases, I guess not all of them need to be handled - at least for now.

Reply:
> I guess when spotting a range with additional spaces like `2015 - 2016` we should still be able to upgrade the year.
> With `2015-16` should we log an error and skip modification?

Yes that sounds perfect!

>  Another thing is I assumed there's only one line with years, so if there are multiple just the first one will get modified. I don't think it's a common case so I didn't support it. There are many negative cases, I guess not all of them need to be handled - at least for now.

Agreed

Reply:
Done. I was actually able to support the `2015 - 16` case. We will now return an error if an attempt to update the years is made, but the current year is not at the end of the updated year list. 
For instance, if `2015 -- 16` is spotted (as in the added test), it would get updated to `2015-2023 -- 16`. Our `_YEARS_LIST` regex would detect it as a range ending in 16, while it should end in 2023, so we will report an error and skip the update.

Reply:
Perfect!